tests/unit/nupic/data/file_record_stream_test.py

```diff
@@ -130,6 +130,110 @@ def testBasic(self):
     s.close()
 
 
+  def testMultipleClasses(self):
+    """Runs FileRecordStream tests with multiple category fields."""
+    filename = _getTempFileName()
+
+    # Write a standard file
+    fields = [('name', 'string', ''),
+              ('timestamp', 'datetime', 'T'),
+              ('integer', 'int', ''),
+              ('real', 'float', ''),
+              ('reset', 'int', 'R'),
+              ('sid', 'string', 'S'),
+              ('categories', 'list', 'C')]
+    fieldNames = ['name', 'timestamp', 'integer', 'real', 'reset', 'sid',
+                  'categories']
+
+    print 'Creating temp file:', filename
+
+    s = FileRecordStream(streamID=filename, write=True, fields=fields)
```

Comment:
`FileRecordStream` can be used as a context manager (`with FileRecordStream(...) as s:`) to ensure that `s.close()` gets called if there is an error.